### ISSUE TITLE
ERROR: KeyError: current_header not found

### ISSUE THREAD

**Issue by sbromberger** [JuliaWeb/Requests.jl#41]
```
julia> z = get("https://github.com")
ERROR: KeyError: current_header not found
 in on_header_value at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:123
 in http_parser_execute at /Users/seth/.julia/v0.4/HttpParser/src/HttpParser.jl:106
 in process_response at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:231
 in do_request at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:565
 in get at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:579
 in get at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:578
```

This is with GnuTLS and the ird/tempfix branch.

**Comment by sbromberger** [comment]
Here are the headers (obtained via GnuTLS `readall` - I broke the lines out at `\r\n`):

```
Server: GitHub.com\r\n
Date: Sun, 15 Feb 2015 22:37:36 GMT\r\n
Content-Type: text/html; charset=utf-8\r\n
Transfer-Encoding: chunked\r\n
Status: 200 OK\r\n
Content-Security-Policy: default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.github.com; style-src 'self' 'unsafe-inline' 'unsafe-eval' assets-cdn.github.com; img-src 'self' data: assets-cdn.github.com identicons.github.com www.google-analytics.com collector.githubapp.com *.githubusercontent.com *.gravatar.com *.wp.com; media-src 'none'; frame-src 'self' render.githubusercontent.com gist.github.com www.youtube.com player.vimeo.com checkout.paypal.com; font-src assets-cdn.github.com; connect-src 'self' ghconduit.com:25035 live.github.com wss://live.github.com uploads.github.com www.google-analytics.com s3.amazonaws.com\r\n
Cache-Control: no-cache\r\n
Vary: X-PJAX\r\n
X-UA-Compatible: IE=Edge,chrome=1\r\n
Set-Cookie: logged_in=no; domain=.github.com; path=/; expires=Thu, 15-Feb-2035 22:37:36 GMT; secure; HttpOnly\r\n
Set-Cookie: _gh_sess=eyJzZXNzaW9uX2lkIjoiMzZkYWY3MmI5YzMzOGVhMDk4NWJkYjRhNDRkN2I3MDMiLCJfY3NyZl90b2tlbiI6InN3VHVNQ2JmSWRBVDk2OGhNcDg4Ni93a05wdjc3R2psdXU0MmxRaGpLNXM9In0%3D--3da1976f55eb551c14a3530bc9b1ab0292856758; path=/; secure; HttpOnly\r\n
X-Request-Id: b6ab0ab016a18302e2bd2f562a8d6e63\r\n
X-Runtime: 0.009179\r\n
X-Rack-Cache: miss\r\n
X-GitHub-Request-Id: 32948C31:202A:26CD3E66:54E11FA6\r\n
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload\r\n
X-Content-Type-Options: nosniff\r\n
X-XSS-Protection: 1; mode=block\r\n
X-Frame-Options: deny\r\n
Vary: Accept-Encoding\r\n
X-Served-By: 4580e595d77515caa5593194518fa90f\r\n
\r\n
a03
\r\n
```

Dumping the dictionary during each call to `on_header_value` gives me this:

```
julia> z = get("https://github.com")
Dict{AbstractString,AbstractString} with 5 entries:
  "current_header"   => "Server"
  "Content-Type"     => "text/html; charset=utf-8"
  "Date"             => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Content-Language" => "en"
  "Server"           => "Julia/0.4.0-dev+3099"
Dict{AbstractString,AbstractString} with 5 entries:
  "current_header"   => "Date"
  "Content-Type"     => "text/html; charset=utf-8"
  "Date"             => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Content-Language" => "en"
  "Server"           => "GitHub.com"
Dict{AbstractString,AbstractString} with 5 entries:
  "current_header"   => "Content-Type"
  "Content-Type"     => "text/html; charset=utf-8"
  "Date"             => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Content-Language" => "en"
  "Server"           => "GitHub.com"
Dict{AbstractString,AbstractString} with 5 entries:
  "current_header"   => "Transfer-Encoding"
  "Content-Type"     => "text/html; charset=utf-8"
  "Date"             => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Content-Language" => "en"
  "Server"           => "GitHub.com"
Dict{AbstractString,AbstractString} with 6 entries:
  "current_header"    => "Status"
  "Transfer-Encoding" => "chunked"
  "Content-Type"      => "text/html; charset=utf-8"
  "Date"              => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Content-Language"  => "en"
  "Server"            => "GitHub.com"
Dict{AbstractString,AbstractString} with 7 entries:
  "current_header"    => "Content-Security-Policy"
  "Transfer-Encoding" => "chunked"
  "Status"            => "200 OK"
  "Content-Type"      => "text/html; charset=utf-8"
  "Date"              => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Content-Language"  => "en"
  "Server"            => "GitHub.com"
Dict{AbstractString,AbstractString} with 8 entries:
  "current_header"          => "Cache-Control"
  "Transfer-Encoding"       => "chunked"
  "Status"                  => "200 OK"
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Content-Language"        => "en"
  "Server"                  => "GitHub.com"
Dict{AbstractString,AbstractString} with 9 entries:
  "current_header"          => "Vary"
  "Transfer-Encoding"       => "chunked"
  "Cache-Control"           => "no-cache"
  "Status"                  => "200 OK"
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Content-Language"        => "en"
  "Server"                  => "GitHub.com"
Dict{AbstractString,AbstractString} with 10 entries:
  "current_header"          => "X-UA-Compatible"
  "Transfer-Encoding"       => "chunked"
  "Cache-Control"           => "no-cache"
  "Status"                  => "200 OK"
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Content-Language"        => "en"
  "Vary"                    => "X-PJAX"
  "Server"                  => "GitHub.com"
Dict{AbstractString,AbstractString} with 11 entries:
  "X-UA-Compatible"         => "IE=Edge,chrome=1"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Transfer-Encoding"       => "chunked"
  "Cache-Control"           => "no-cache"
  "Status"                  => "200 OK"
  "Server"                  => "GitHub.com"
  "current_header"          => "Set-Cookie"
  "Vary"                    => "X-PJAX"
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Content-Language"        => "en"
Dict{AbstractString,AbstractString} with 12 entries:
  "X-UA-Compatible"         => "IE=Edge,chrome=1"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Transfer-Encoding"       => "chunked"
  "Cache-Control"           => "no-cache"
  "Status"                  => "200 OK"
  "Server"                  => "GitHub.com"
  "current_header"          => "Set-Cookie"
  "Vary"                    => "X-PJAX"
  "Set-Cookie"              => "logged_in=no; domain=.github.com; path=/; expires=Thu, 15-Feb-2035 22:35:25 GMT; secure; HttpOnly"
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Content-Language"        => "en"
Dict{AbstractString,AbstractString} with 12 entries:
  "X-UA-Compatible"         => "IE=Edge,chrome=1"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Transfer-Encoding"       => "chunked"
  "Cache-Control"           => "no-cache"
  "Status"                  => "200 OK"
  "Server"                  => "GitHub.com"
  "current_header"          => "X-Request-Id"
  "Vary"                    => "X-PJAX"
  "Set-Cookie"              => "_gh_sess=eyJzZXNzaW9uX2lkIjoiNzc2MjcwZGFjOGU4Mzg5M2ZjYWJhNTIyNDYzYzE2MGYiLCJfY3NyZl90b2tlbiI6InJKS3d…
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Content-Language"        => "en"
Dict{AbstractString,AbstractString} with 13 entries:
  "X-UA-Compatible"         => "IE=Edge,chrome=1"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Transfer-Encoding"       => "chunked"
  "Cache-Control"           => "no-cache"
  "Status"                  => "200 OK"
  "X-Request-Id"            => "e8ff640f8daaf972d14f61a018a2a241"
  "Server"                  => "GitHub.com"
  "current_header"          => "X-Runtime"
  "Vary"                    => "X-PJAX"
  "Set-Cookie"              => "_gh_sess=eyJzZXNzaW9uX2lkIjoiNzc2MjcwZGFjOGU4Mzg5M2ZjYWJhNTIyNDYzYzE2MGYiLCJfY3NyZl90b2tlbiI6InJKS3d…
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Content-Language"        => "en"
Dict{AbstractString,AbstractString} with 14 entries:
  "X-UA-Compatible"         => "IE=Edge,chrome=1"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Transfer-Encoding"       => "chunked"
  "Cache-Control"           => "no-cache"
  "Status"                  => "200 OK"
  "X-Request-Id"            => "e8ff640f8daaf972d14f61a018a2a241"
  "X-Runtime"               => "0.007431"
  "Server"                  => "GitHub.com"
  "current_header"          => "X-Rack-Cache"
  "Vary"                    => "X-PJAX"
  "Set-Cookie"              => "_gh_sess=eyJzZXNzaW9uX2lkIjoiNzc2MjcwZGFjOGU4Mzg5M2ZjYWJhNTIyNDYzYzE2MGYiLCJfY3NyZl90b2tlbiI6InJKS3d…
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Content-Language"        => "en"
Dict{AbstractString,AbstractString} with 15 entries:
  "X-UA-Compatible"         => "IE=Edge,chrome=1"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Transfer-Encoding"       => "chunked"
  "Cache-Control"           => "no-cache"
  "Status"                  => "200 OK"
  "X-Request-Id"            => "e8ff640f8daaf972d14f61a018a2a241"
  "X-Runtime"               => "0.007431"
  "Server"                  => "GitHub.com"
  "current_header"          => "X-GitHub-Request-Id"
  "Vary"                    => "X-PJAX"
  "Set-Cookie"              => "_gh_sess=eyJzZXNzaW9uX2lkIjoiNzc2MjcwZGFjOGU4Mzg5M2ZjYWJhNTIyNDYzYzE2MGYiLCJfY3NyZl90b2tlbiI6InJKS3d…
  "X-Rack-Cache"            => "miss"
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Content-Language"        => "en"
Dict{AbstractString,AbstractString} with 15 entries:
  "X-UA-Compatible"         => "IE=Edge,chrome=1"
  "X-GitHub-Request-Id"     => "32948"
  "Date"                    => "Sun, 15 Feb 2015 22:35:25 GMT"
  "Transfer-Encoding"       => "chunked"
  "Cache-Control"           => "no-cache"
  "Status"                  => "200 OK"
  "X-Request-Id"            => "e8ff640f8daaf972d14f61a018a2a241"
  "X-Runtime"               => "0.007431"
  "Server"                  => "GitHub.com"
  "Vary"                    => "X-PJAX"
  "Set-Cookie"              => "_gh_sess=eyJzZXNzaW9uX2lkIjoiNzc2MjcwZGFjOGU4Mzg5M2ZjYWJhNTIyNDYzYzE2MGYiLCJfY3NyZl90b2tlbiI6InJKS3d…
  "X-Rack-Cache"            => "miss"
  "Content-Security-Policy" => "default-src *; script-src assets-cdn.github.com collector-cdn.github.com; object-src assets-cdn.gith…
  "Content-Type"            => "text/html; charset=utf-8"
  "Content-Language"        => "en"
ERROR: KeyError: current_header not found
 in on_header_value at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:124
 in http_parser_execute at /Users/seth/.julia/v0.4/HttpParser/src/HttpParser.jl:106
 in process_response at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:232
 in do_request at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:566
 in get at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:580
 in get at /Users/seth/.julia/v0.4/Requests/src/Requests.jl:579
```

**Comment by sbromberger** [comment]
This is timing related. If I insert a sleep(0.001) after `data = readavailable(stream)` in `process_response`, there is no issue. Maddening and a sign of fragility, I think.

**Comment by sbromberger** [comment]
Closed via #43.
